src/server/middleware/blob/head_blob.go

```diff
@@ -0,0 +1,58 @@
+package blob
+
+import (
+	"context"
+	"fmt"
+	"github.com/goharbor/harbor/src/controller/blob"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	blob_models "github.com/goharbor/harbor/src/pkg/blob/models"
+	serror "github.com/goharbor/harbor/src/server/error"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"net/http"
+)
+
+// HeadBlobMiddleware intercept the head blob request
+func HeadBlobMiddleware() func(http.Handler) http.Handler {
+	return middleware.New(func(rw http.ResponseWriter, req *http.Request, next http.Handler) {
+		if err := handleHead(req); err != nil {
+			serror.SendError(rw, err)
+			return
+		}
```

Comment:
The code of handleHead and handleBlob can move to here.

Reply:
updated.